neuron et al: set variables better?

matz-e opened the pull request:
Alternative to #690

matz-e committed:
neuron et al: set variables better?

matz-e commented:
@pramodk remove fake version then ok?

ohm314 commented on var/spack/repos/builtin/packages/sim-model/package.py:
```diff
@@ -67,7 +67,7 @@ def _build_mods(self, mods_location, link_flag='', include_flag='', corenrn_mods
 
         include_flag += ' -DENABLE_TAU_PROFILER' if '+profile' in self.spec else ''
-        output_dir = os.path.basename(self.neuron_archdir)
+        output_dir = os.path.basename(self.spec.neuron_archdir)
```
```suggestion
        output_dir = os.path.basename(spec.neuron_archdir)
```
why not like this? as done above...

matz-e replied:
because copy-pasta gone wrong?

matz-e committed:
Update var/spack/repos/builtin/packages/sim-model/package.py

Co-Authored-By: Omar Awile <omar.awile@epfl.ch>

matz-e commented on var/spack/repos/builtin/packages/sim-model/package.py:
```diff
@@ -67,7 +67,7 @@ def _build_mods(self, mods_location, link_flag='', include_flag='', corenrn_mods
 
         include_flag += ' -DENABLE_TAU_PROFILER' if '+profile' in self.spec else ''
-        output_dir = os.path.basename(self.neuron_archdir)
+        output_dir = os.path.basename(spec.neuron_archdir)
```
```suggestion
        output_dir = os.path.basename(self.spec.neuron_archdir)
```
Actually, `spec` is not available in this function.

matz-e committed:
Update var/spack/repos/builtin/packages/sim-model/package.py

iomaganaris commented:
@matz-e I hope it's okay I took over the PR and rebased to `develop` 😇

iomaganaris committed:
Fixed formatting according to flake8

iomaganaris committed:
Attempt to fix issue when build neurodamus-core with dev-build

There is an issue when building for example neurodamus-neocortex.
During the installation of neurodamus-core it tries to delete
mod/VecStim.mod (probably) from the neurodamus-neocortex source folder,
not finding it and exiting with error.

matz-e commented:
@iomaganaris does that mean you own it now? Let me review tomorrow, Travis also has complaints already :-D

_no complaints from me here 🙃 _

iomaganaris committed:
Revert "Attempt to fix issue when build neurodamus-core with dev-build"

This reverts commit 8bbc1f519cbea0991f2e136d23196f7451918083.

iomaganaris commented:
> @iomaganaris does that mean you own it now? Let me review tomorrow, Travis also has complaints already :-D
> 
> _no complaints from me here 🙃 _

I think that it was a bad choice to touch the old `neurodamus` package. Travis doesn't seem to like it 😛 
I will also check what is the issue tomorrow.

iomaganaris committed:
Remove old neurodamus recipe as in develop

ohm314 reviewed (approved):
I like it!

matz-e reviewed (commented):
Me too, but I can't approve :-/
